File: Lib/Telerik/Examples/RichTextEditor/SyntaxHighlight/Parsers/SyntaxPicker.cs
using System.Collections.Generic;

namespace RichTextEditor.Parsers
{
	internal class SyntaxPicker
	{
		private List<SyntaxParser> parsers = new List<SyntaxParser>();

		internal SyntaxPicker()
		{
			this.AddParser(new CSharpSyntaxParser());
			this.AddParser(new XamlSyntaxParser());
		}

		internal void AddParser(SyntaxParser parser)
		{
			this.parsers.Add(parser);
		}

		internal SyntaxParser FindParserByExtension(string extension)
		{
			foreach (SyntaxParser parser in this.parsers)
			{
				if (parser.GetExtension == extension)
					return parser;
			}
			return null;
		}
	}
}

File: Lib/Telerik/Examples/RichTextEditor/SyntaxHighlight/Parsers/SyntaxPicker.vb
Imports Microsoft.VisualBasic
Imports System.Collections.Generic

Namespace RichTextEditor.Parsers
	Friend Class SyntaxPicker
		Private parsers As List(Of SyntaxParser) = New List(Of SyntaxParser)()

		Friend Sub New()
			Me.AddParser(New CSharpSyntaxParser())
			Me.AddParser(New XamlSyntaxParser())
		End Sub

		Friend Sub AddParser(ByVal parser As SyntaxParser)
			Me.parsers.Add(parser)
		End Sub

		Friend Function FindParserByExtension(ByVal extension As String) As SyntaxParser
			For Each parser As SyntaxParser In Me.parsers
				If parser.GetExtension = extension Then
					Return parser
				End If
			Next parser
			Return Nothing
		End Function
	End Class
End Namespace
